src/contrib/cluster/Akka.Cluster.Tools/Client/Serialization/ClusterClientMessageSerializer.cs

```diff
@@ -112,47 +109,23 @@ public override string Manifest(object o)
             if (o is ClusterReceptionist.Heartbeat) return HeartbeatManifest;
             if (o is ClusterReceptionist.HeartbeatRsp) return HeartbeatRspManifest;
 
-            throw new ArgumentException($"Can't serialize object of type [{o.GetType()}] in [{GetType()}]");
+            throw new ArgumentException($"Can't serialize object of type [{o.GetType()}] in [{nameof(ClusterClientMessageSerializer)}]");
         }
 
-        private byte[] Compress(IMessageLite message)
+        private byte[] ContactsToProto(ClusterReceptionist.Contacts message)
         {
-            using (var bos = new MemoryStream(BufferSize))
-            using (var gzipStream = new GZipStream(bos, CompressionMode.Compress))
+            var protoMessage = new Proto.Msg.Contacts();
+            foreach (var contactPoint in message.ContactPoints)
             {
-                message.WriteTo(gzipStream);
-                gzipStream.Dispose();
-                return bos.ToArray();
+                protoMessage.ContactPoints.Add(contactPoint);
             }
-        }
-
-        private byte[] Decompress(byte[] bytes)
```

Comment:
I'm not sure that we should have compress/decompress logic inside serializers. I think it should be done in the remote transport. But DotNetty does not support it at the moment